### ISSUE TITLE
ACD rclone failing to upload and endless restarting

### ISSUE THREAD

**Issue by mdkberry** [rclone/rclone#1113]
> What is your rclone version (eg output from `rclone -V`)
rclone: Version "v1.35" 
> Which OS you are using and how many bits (eg Windows 7, 64 bit)
win pro 7 32 bit
> Which cloud storage system are you using? (eg Google Drive)
ACD
> The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
starting with parameters ["rclone" "copy" "C:\\Users\\admin\\Desktop\\To_Cloud" "acd:Backups\\" "--log-file=Log.txt" "--acd-upload-wait-per-gb" "30m" "--verbose"]

I am going out a Mikrotik Rb2011 router which I have access to and has no traffic shaping on it. 

Problem:
I have been running rclone for over a month trying to upload my backups to ACD with all zip files at 4.4GB in size, and it was working fine but it has deteriorated and started failing over the last month, and the last 2 weeks has stopped uploading altogether. 

I have moved house since and started a new internet service ,no longer ADSL but fibre, and same provider iiNet, in Australia.  This time with 25down /1mbps up, but the actual upload fluctuates between 900k and 1400k when I monitor it, or use speedtest.net. On this I am still getting mostly failures but maybe 1 in 6 upload attempts work and will run for 8 to 18 hours to upload a single 4.4gb zip file. 

I have searched for solution and found nothing that works yet. I contacted Amazon and they checked my service their end, then said it is rclone at fault, of course. 

here are some of the errors. any ideas what I can do to sort this I have tried --bwlimit 900k but it failed constantly, so I am back trying without anything but verbose on it and the upload-wait-per-gb setting at 30m which helped a bit. I am currently only moving 1 file at a time in order to isolate the issue and try to get something uploaded. 

at this point I am 300GB short of my upload target and pulling my hair out tbh. will have to ditch amazon and start elsewhere if i can't resolve this issue.

here is one error and right after it the upload starts over...

_2017/02/08 12:04:51 Mark_Masters.zip.003: Upload error detected but didn't finish upload: HTTP code 504: "504 GATEWAY_TIMEOUT": no response body ("504 GATEWAY_TIMEOUT")
2017/02/08 12:04:51 pacer: Rate limited, sleeping for 947.77941ms (1 consecutive low level retries)
2017/02/08 12:04:51 pacer: low level retry 1/1 (error HTTP code 504: "504 GATEWAY_TIMEOUT": no response body)
2017/02/08 12:04:51 Mark_Masters.zip.003: Received error: HTTP code 504: "504 GATEWAY_TIMEOUT": no response body - low level retry 1/10
2017/02/08 12:04:52 pacer: Resetting sleep to minimum 20ms on success_

but the more common error I am seeing is this one and again it starts over right afterwards. 

_2017/02/08 17:11:48 Mark_Masters.zip.003: Upload error detected but didn't finish upload: Post https://content-na.drive.amazonaws.com/cdproxy/nodes?suppress=deduplication: write tcp 192.168.88.238:55296->52.1.75.19:443: wsasend: An existing connection was forcibly closed by the remote host. ("HTTP status UNKNOWN")
2017/02/08 17:11:48 pacer: Rate limited, sleeping for 82.153551ms (1 consecutive low level retries)
2017/02/08 17:11:48 pacer: low level retry 1/1 (error Post https://content-na.drive.amazonaws.com/cdproxy/nodes?suppress=deduplication: write tcp 192.168.88.238:55296->52.1.75.19:443: wsasend: An existing connection was forcibly closed by the remote host.)
2017/02/08 17:11:48 Mark_Masters.zip.003: Received error: Post https://content-na.drive.amazonaws.com/cdproxy/nodes?suppress=deduplication: write tcp 192.168.88.238:55296->52.1.75.19:443: wsasend: An existing connection was forcibly closed by the remote host. - low level retry 2/10
2017/02/08 17:11:49 pacer: Resetting sleep to minimum 20ms on success
2017/02/08 17:11:50 Mark_Masters.zip.003: Upload error detected but didn't finish upload: HTTP code 429: "429 Too Many Requests": response body: "{\"logref\":\"e1f66a94-edcd-11e6-a160-c1e1ae76cd6a\",\"message\":\"Rate exceeded\",\"code\":\"\"}" ("429 Too Many Requests")
2017/02/08 17:11:50 pacer: Rate limited, sleeping for 666.145821ms (1 consecutive low level retries)
2017/02/08 17:11:50 pacer: low level retry 1/1 (error HTTP code 429: "429 Too Many Requests": response body: "{\"logref\":\"e1f66a94-edcd-11e6-a160-c1e1ae76cd6a\",\"message\":\"Rate exceeded\",\"code\":\"\"}")
2017/02/08 17:11:50 Mark_Masters.zip.003: Received error: HTTP code 429: "429 Too Many Requests": response body: "{\"logref\":\"e1f66a94-edcd-11e6-a160-c1e1ae76cd6a\",\"message\":\"Rate exceeded\",\"code\":\"\"}" - low level retry 3/10
2017/02/08 17:11:51 pacer: Resetting sleep to minimum 20ms on success
2017/02/08 17:12:22 
Transferred:   2.496 GBytes (115.071 kBytes/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:   6h19m2.5s
Transferring:
 *_

**Comment by ajkis** [comment]
Can you try with 
`rclone copy C:\Users\admin\Desktop\To_Cloud acd:Backups\  --checksum --no-traverse --transfers=1 --checkers=1  --retries=10 --low-level-retries=20 --contimeout 10m --acd-upload-wait-per-gb 30m --verbose --log-file=Log.txt`

**Comment by ncw** [comment]
> 2017/02/08 17:11:48 Mark_Masters.zip.003: Upload error detected but didn't finish upload: Post https://content-na.drive.amazonaws.com/cdproxy/nodes?suppress=deduplication: write tcp 192.168.88.238:55296->52.1.75.19:443: wsasend: An existing connection was forcibly closed by the remote host. ("HTTP status UNKNOWN")

That looks like it could be your ISP, your router or your firewall closing connections.

That probably explains why you are having problems.

Check the firmware on your router is probably the most productive line of attack (I've had lots of reports from rclone users of routers flaking out when rclone pummels them with data!).

**Comment by mdkberry** [comment]
thanks guys. not working still. most annoying is it usually goes as far as 98 or 99% before failing as can be seen here. I used your suggestion on this go ajkis unfortunately it hasnt helped. I dont believe it is the router as it was working for over a month without this happeneing and I have only changed the isp settings so thinking maybe isp maybe modem but I really saw deterioration happen before moving that could only be ACD end even though they deny it, nothing else had changed in the weeks it started. 

any other ideas to resolve this would really help i see no choice but to look at google next and its a long wasted journey for me to get this far and it all fail.

anyway here is todays error and restart point after trying this ...

_**rclone copy C:\Users\admin\Desktop\To_Cloud acd:/Backups --checksum --no-traverse --transfers=1 --checkers=1 --retries=10 --low-level-retries=20 --contimeout 10m --acd-upload-wait-per-gb 30m --verbose --log-file=Log.txt**_


 _*                          Mark_Masters.zip.005: 99% done, 13 Bytes/s, ETA: 894h1m0s

2017/02/11 07:28:14 Mark_Masters.zip.005: Upload error detected but didn't finish upload: Post https://content-na.drive.amazonaws.com/cdproxy/nodes?suppress=deduplication: write tcp 192.168.88.238:50478->54.172.239.173:443: wsasend: An existing connection was forcibly closed by the remote host. ("HTTP status UNKNOWN")
2017/02/11 07:28:14 pacer: Rate limited, sleeping for 947.77941ms (1 consecutive low level retries)
2017/02/11 07:28:14 pacer: low level retry 1/1 (error Post https://content-na.drive.amazonaws.com/cdproxy/nodes?suppress=deduplication: write tcp 192.168.88.238:50478->54.172.239.173:443: wsasend: An existing connection was forcibly closed by the remote host.)
2017/02/11 07:28:14 Mark_Masters.zip.005: Received error: Post https://content-na.drive.amazonaws.com/cdproxy/nodes?suppress=deduplication: write tcp 192.168.88.238:50478->54.172.239.173:443: wsasend: An existing connection was forcibly closed by the remote host. - low level retry 1/20
2017/02/11 07:28:16 pacer: Resetting sleep to minimum 20ms on success
2017/02/11 07:29:14 
Transferred:   4.342 GBytes (116.914 kBytes/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:    10h49m4s
Transferring:
 *                          Mark_Masters.zip.005:  0% done, 121.259 kBytes/s, ETA: 10h29m20s_

**Comment by ajkis** [comment]
Can you download PingInfoView http://www.nirsoft.net/utils/multiple_ping_tool.html ( no installation just executable ) and run it and set it to ping:
your local gateway 
your ISP gateway ( check on your router what it is )
8.8.8.8 ( google dns )
+ 1 more public internet IP 

With it you will be able to debug whats going on, it could be that you are loosing connection to your router, the tool will show you exactly how much packet loss you have and how many dropped pings
http://puu.sh/tXtXm/b63e3145fe.png

**Comment by mdkberry** [comment]
cool. have started running that and already getting some 'time out' ping failures to both the 8.8.8.8 and the secondary public ip address I added which is 139.130.4.5 (Telstra) the fails were at different times and so far they are both at 0.6% failed. 
all other pings 100% success so far. its only been 10 minutes.

I am going to add a third public ip in and try to find something reliable to confirm this is a clear issue or not

**Comment by mdkberry** [comment]
ok looks like ISP or maybe between my router and the outside world. the ping to the router is the only thing that hasnt failed so far with timeouts. meanwhile the rclone upload is running but I am guessing this is likely cause I will wait to see what I get when the rclone upload next fails. thankyou for your help I think my next conversation will be with iiNet the ISP.

**Comment by mdkberry** [comment]
40 minutes later and I am up to x8 ping failures to the ISP gateway, x7 to 8.8.8.8, x3 to the other public ip's, and still 0 failures to my router. I am doing this while the rclone upload is running.

so as an additional test to confirm it isnt my 'pipes' being full of rclone traffic causing the fails, for the duration I have kept a VPN link up from another machine to an outside VPN, this so far has not been dropped.

**Comment by mdkberry** [comment]
just an update: my iinet ISP service ping failures to the gateway or anything beyond hovered around 3-4% failure and no failures to my router. 

I have been running ajkis rclone suggestion and it has helped. I get failures but am getting them uploaded within x3 attempts so it is better than what I was experiencing and I can work with it for now.

I will take it up with my ISP next and see if I can improve the ping failures. thanks for your help this at least helped me focus in on the problem area and I am limping the uploads in which is an improvement.

If I get any revelations from the ISP I will post them here.

**Comment by felixbuenemann** [comment]
My guess is that there is a too low idle connection timeout on your router or at your ISP.

After the file is uploaded to ACD it just sits there for up to two minutes, while the ACD Backend checks the file to generate the checksum (and maybe video preview). After this time the backend returns a 408 Error (Request Timed Out) which then causes rclone to enter the waiting-for-upload loop.

There are also occasionally 504 or 429 errors, which usually mean file will appear after waiting too, although the 429 (rate limited) error can shadow fatal errors.

Sometimes the api returns a 500 or 400 error, which in my appearance means the upload failed and must be retried.

If the connection is simply closed this sounds like it is happening either on your router (check the firewall settings related to [connection tracking](https://wiki.mikrotik.com/wiki/Manual:IP/Firewall/Connection_tracking).

If the timeouts on your router are large enough it might also be a NAT gateway or transparent proxy at your ISP (many providers nowadays use ipv6 with nat-gateways for ipv4), there are some reports on ssh/telnet session timeout with iinet [here](http://forums.whirlpool.net.au/archive/69277).

One thing you could try is to copy with all retries disabled and manually check after letzt say 10 to 15 minutes in the amazon drive web ui wether your files appeared.

```
rclone copy \
  --transfers 1 \
  --low-level-retries 1 \
  --retries 1 \
  --size-only \
  --acd-upload-wait-per-gb 0 \
  --max-size 50000M \
  src_folder acd:dst_folder
```

This could work, because the transfer has likely completed, but you get an error because your connection is closed while waiting for the response from the ACD backend.

Note that this command will intentionally not retry on error, so you need to restart it periodically to retry the failed transfers.

**Comment by mdkberry** [comment]
thanks that is helpful. I tried all sorts of methods and am currently working with 

 _rclone: Version "v1.35"  ["rclone" "copy" "C:\\Users\\admin\\Desktop\\To_Cloud" "acd:Backups\\" "--checksum" "--no-traverse" "--transfers=1" "--checkers=1" "--retries=10" "--low-level-retries=20" "--contimeout" "10m" "--acd-upload-wait-per-gb" "30m" "--verbose" "--log-file=Log.txt"]_

this works ok. but I had to reduce the file sizes I was copying to 700MB , where originally I was working fine with 4.4GB files without a problem this is the sweet point where it gets enough time to upload without a failure occuring. 

I still get failures but the success rate is adequate for me with 700Mb and it takes about 1 hour.

Interestingly I had a period of time where my Upload speed was increased with the ISP iinet to 25 download and 2mb or more upload. during this period it showed no failures and I managed to get it pushed through much quicker. I am now thinking that 1mb upload speed limit is additionally constrictive to the entire process. 2mb is probably a minimum and may see the end of the errors I have been experiencing. 

But like I said it is working currently with 700mb size files and taking about 1.5 hours for each to get completed. Some fail at 99% and the start over but on the whole it is working. I can live with it. Just sucks I couldt upload larger files but hey-ho.

thanks everyone for your help
